src/client/interpreter/locators/services/baseVirtualEnvService.ts

```
@@ -0,0 +1,88 @@
+        super(name, serviceContainer, cachePerWorkspace);
+        this.virtualEnvMgr = serviceContainer.get<IVirtualEnvironmentManager>(IVirtualEnvironmentManager);
+        this.versionProvider = serviceContainer.get<IInterpreterVersionService>(IInterpreterVersionService);
+        this.fileSystem = serviceContainer.get<IFileSystem>(IFileSystem);
+    }
+    // tslint:disable-next-line:no-empty
+    public dispose() { }
+    protected getInterpretersImplementation(resource?: Uri): Promise<PythonInterpreter[]> {
+        return this.suggestionsFromKnownVenvs(resource);
+    }
+    private async suggestionsFromKnownVenvs(resource?: Uri) {
+        const searchPaths = this.searchPathsProvider.getSearchPaths(resource);
+        return Promise.all(searchPaths.map(dir => this.lookForInterpretersInVenvs(dir)))
+            .then(listOfInterpreters => _.flatten(listOfInterpreters));
+    }
+    private async lookForInterpretersInVenvs(pathToCheck: string) {
+        return this.fileSystem.getSubDirectoriesAsync(pathToCheck)
+            .then(subDirs => Promise.all(this.getProspectiveDirectoriesForLookup(subDirs)))
+            .then(dirs => dirs.filter(dir => dir.length > 0))
+            .then(dirs => Promise.all(dirs.map(lookForInterpretersInDirectory)))
+            .then(pathsWithInterpreters => _.flatten(pathsWithInterpreters))
+            .then(interpreters => Promise.all(interpreters.map(interpreter => this.getVirtualEnvDetails(interpreter))))
+            .catch((err) => {
+                console.error('Python Extension (lookForInterpretersInVenvs):', err);
+                // Ignore exceptions.
+                return [] as PythonInterpreter[];
+            });
+    }
+    private getProspectiveDirectoriesForLookup(subDirs: string[]) {
+        const isWindows = this.serviceContainer.get<IPlatformService>(IPlatformService).isWindows;
+        const dirToLookFor = isWindows ? 'SCRIPTS' : 'bin';
+        return subDirs.map(subDir =>
```

You can probably just await and drop try-catch since the block is just file system operations and if they fail, lots of other things will get broken anyway

---

I need the try catch, because I want to ignore errors.